### PR TITLE
feat(List Patches): Added Page to List Patches.

### PR DESCRIPTION
**Pull request by deeptea22**
### Functionality:
A page to display all the patches.

### Solution:
Added 3 components to list patches.

### Risk level:
- [ ] high 
- [x] medium
- [ ] low

### How to test:
ng serve
firebase emulators:start

### CODE REVIEW

**Review by shivamgoyal7 — APPROVED** [review]
Ship it

File: src/app/list-patches/edit-patch/edit-patch.component.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  async editPage(){
+    this.enableLoader = true;
+    try{            
+      const patchDB = this.db.collection('Patches');
+      patchDB.doc(this.patch).update({Name: this.editPatch.Name});
+      patchDB.doc(this.patch).update({Description: this.editPatch.Description});
+      this.enableLoader = false;
+      this.showClose = true;
```
Comment (viv9k) [inline src/app/list-patches/edit-patch/edit-patch.component.ts:60]: Inappropriate way to call api, This doesn't explain things properly

File: src/app/list-patches/edit-patch/edit-patch.component.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  async generateChanges() {    
+    if (this.prevVal[1] != this.newVal[1])
+      this.changedData = this.changedData + " description,";
+    if (this.prevVal[0] != this.newVal[0])
+      this.changedData = this.changedData + " name,";
+    if (this.changedData != "")
+      this.changedData = "Edited-" + this.changedData;
+    this.changedData = this.changedData.substring(0, this.changedData.length - 1) + "."
+    console.log(this.changedData);
+  }
```
Comment (viv9k) [inline src/app/list-patches/edit-patch/edit-patch.component.ts:51]: This  doesn't explain why we have such an algorithm

File: src/app/list-patches/edit-patch/edit-patch.component.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  changedData: string = ""
+  enableLoader: boolean = false;
+  constructor( public db: AngularFirestore,private patchService: PatchService) { }
+
+  ngOnInit(): void {  
+    this.patchService.getPatchData(this.patch).subscribe(data => {     
+      this.initialPatch = data;
+      this.prevVal = [this.initialPatch.Name,this.initialPatch.Description];      
+      this.editPatch = this.initialPatch;
+    });
```
Comment (viv9k) [inline src/app/list-patches/edit-patch/edit-patch.component.ts:29]: Not a right way to handle data.

Comment (viv9k) [reply src/app/list-patches/edit-patch/edit-patch.component.ts:29]: Check other places how it's done.

File: src/app/list-patches/patch-cards/patch-cards.component.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    this.editPatchEnabled = true;
+    this.patchService.getPatchData(name).subscribe(data => {      
+      this.patch = data;     
+    });    
+  }
+
+  editPatches(name: string) { 
+    this.EditedName = name;
+    this.editPatchEnabled = true;
+  }
```
Comment (viv9k) [inline src/app/list-patches/patch-cards/patch-cards.component.ts:48]: The edit patch isn't that good

**Review by viv9k — CHANGES_REQUESTED** [review]
please do the required.

**Pull request closed without merging**